### PR TITLE
feat(bench): add string searching bench

### PR DESCRIPTION
**Pull request by Fdawgs**
Inspired by this beast: https://github.com/fastify/send/pull/39

### CODE REVIEW

File: bench/string-searching.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+suite.add('Using includes', function () {
+  text.includes('application/json')
+})
+.add('Using indexof', function () {
+  text.indexOf('application/json') !== -1
+})
+.add('Using RegExp.test', function () {
+  /application\/json/.test(text)
+})
```
Comment (H4ad) [inline bench/string-searching.js:21]: I think you can include another one with RegExp cached.

Comment (Fdawgs) [reply bench/string-searching.js:21]: You've lost me @H4ad, mind suggesting an example?

File: bench/string-searching.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+const text = 'text/html,application/xhtml+xml,application/xml;application/json;q=0.9,image/avif,image/webp,*/*;q=0.8'
+
+suite.add('Using includes', function () {
+  text.includes('application/json')
+})
+.add('Using indexof', function () {
+  text.indexOf('application/json') !== -1
+})
+.add('Using RegExp.test', function () {
```
Comment (RafaelGSS) [inline bench/string-searching.js:19]: Maybe include `new RegExp(...).test(text)` too?

File: bench/string-searching.js (hunk 1 above)
Comment (H4ad) [reply bench/string-searching.js:21]:
```js
const testJson = /application\/json/;

.add('Usingcached RegExp.test', function () {
  testJson.test(text)
})
```

File: bench/string-searching.js (hunk 2 above)
Comment (Fdawgs) [reply bench/string-searching.js:20]: Added!

File: bench/string-searching.js (hunk 1 above)
Comment (Fdawgs) [reply bench/string-searching.js:22]: Added!

**Review by RafaelGSS — APPROVED** [review]

**Pull request merged**